src/codewhisperer/commands/startTransformByQ.ts

```diff
@@ -153,196 +321,94 @@ export async function startTransformByQ() {
-                sessionJobHistory,
-                convertDateToTimestamp(startTime),
-                transformByQState.getProjectName(),
-                transformByQState.getStatus(),
-                convertToTimeString(durationInMs),
-                transformByQState.getJobId()
-            )
-        }
-        if (transformByQState.isSucceeded()) {
-            void vscode.window.showInformationMessage(CodeWhispererConstants.transformByQCompletedMessage)
-        } else if (transformByQState.isPartiallySucceeded()) {
-            void vscode.window.showInformationMessage(CodeWhispererConstants.transformByQPartiallyCompletedMessage)
+            await stopJob(transformByQState.getJobId())
+            void vscode.window.showErrorMessage(CodeWhispererConstants.transformByQCancelledMessage)
+        } catch {
+            void vscode.window.showErrorMessage(CodeWhispererConstants.errorStoppingJobMessage)
         }
-
-        if (payloadFileName !== '') {
-            fs.rmSync(payloadFileName, { recursive: true, force: true }) // delete ZIP if it exists
+    } else {
+        transformByQState.setToFailed()
+        codeTransformTelemetryState.setResultStatus('JobFailed')
+        let displayedErrorMessage =
+            transformByQState.getJobFailureErrorMessage() || CodeWhispererConstants.transformByQFailedMessage
+        if (transformByQState.getJobFailureReason() !== '') {
+            displayedErrorMessage += `: ${transformByQState.getJobFailureReason()}`
         }
+        void vscode.window.showErrorMessage(displayedErrorMessage)
+    }
+    if (sessionPlanProgress['uploadCode'] !== StepProgress.Succeeded) {
+        sessionPlanProgress['uploadCode'] = StepProgress.Failed
+    }
+    if (sessionPlanProgress['buildCode'] !== StepProgress.Succeeded) {
+        sessionPlanProgress['buildCode'] = StepProgress.Failed
+    }
+    if (sessionPlanProgress['transformCode'] !== StepProgress.Succeeded) {
+        sessionPlanProgress['transformCode'] = StepProgress.Failed
     }
+    if (sessionPlanProgress['returnCode'] !== StepProgress.Succeeded) {
+        sessionPlanProgress['returnCode'] = StepProgress.Failed
+    }
+    // Log error to VSCode logs
+    getLogger().error('Amazon Q Code Transform', error)
+}
+
+export async function cleanupTransformJob(intervalId: NodeJS.Timeout | undefined) {
     clearInterval(intervalId)
-    transformByQState.setToNotStarted() // so that the "Transform by Q" button resets
-    transformByQState.setPolledJobStatus('') // reset polled job status too
+    transformByQState.setJobDefaults()
     await vscode.commands.executeCommand('setContext', 'gumby.isStopButtonAvailable', false)
     await vscode.commands.executeCommand('aws.amazonq.refresh')
-    void vscode.commands.executeCommand('aws.amazonq.showPlanProgressInHub', startTime.getTime())
+    void vscode.commands.executeCommand('aws.amazonq.showPlanProgressInHub', codeTransformTelemetryState.getStartTime())
```

Comment:
I see lots of executeCommand() usage in this PR and related code. Commands are for users, except in rare cases where a module can't get a reference to a function. In most cases, it's strongly advised to call functions directly, that results in a more useful stacktrace and type-checking.

Reply:
From what I'm aware of, some of these are using this execute some updates in our around the bottom panel webview. 
Where is a good example for the best practice on this?

The refresh commands scattered throughout are to update the native VSCode tree UI with status as well. I don't think there is another way to do this?

I think your comments are important and we should definitely continue the discussion here or elsewhere. This refactor is more for just semantic and structural organization of our mono function.

![Screenshot 2024-01-12 at 8 08 56 AM](https://github.com/aws/aws-toolkit-vscode/assets/38665452/e0f8dc7b-07c5-4a87-9b86-a11c34b8626f)

<img width="943" alt="Screenshot 2024-01-12 at 8 10 06 AM" src="https://github.com/aws/aws-toolkit-vscode/assets/38665452/9fdc2544-2396-4f16-a87d-70d2dc8fbf96" />

Reply:
> Where is a good example for the best practice on this?

The best practice is to just call functions instead of using `executeCommand`.

The commands are implemented by functions. Import those functions and call them, instead of using `executeCommand`.

> The refresh commands scattered throughout are to update the native VSCode tree UI with status as well. I don't think there is another way to do this?

For now we can ignore the "refresh" commands since that is a hack that carried over from codewhisperer. But for most other commands there is no reason to use `executeCommand`.